src/storage/src/hummock/utils.rs

```diff
@@ -75,28 +76,37 @@ pub fn validate_table_key_range(levels: &[Level]) -> HummockResult<()> {
     Ok(())
 }
 
-pub fn filter_single_sst<R, B>(info: &SstableInfo, key_range: &R) -> bool
+pub fn filter_single_sst<R, B>(info: &SstableInfo, table_id: TableId, key_range: &R) -> bool
 where
     R: RangeBounds<B>,
     B: AsRef<[u8]>,
 {
     let table_range = info.key_range.as_ref().unwrap();
     let table_start = user_key(table_range.left.as_slice());
     let table_end = user_key(table_range.right.as_slice());
+    #[cfg(any(test, feature = "test"))]
+    if table_id.table_id() == 0 {
+        return range_overlap(key_range, table_start, table_end);
+    }
     range_overlap(key_range, table_start, table_end)
+        && info
+            .get_table_ids()
+            .binary_search(&table_id.table_id())
```

Comment:
We assume table_ids in SstableInfo is sorted. Do we check it somewhere? If not, we should add a debug_assert in `commit_epoch` and `report_compact_task`.

Reply:
> We assume table_ids in SstableInfo is sorted. Do we check it somewhere? If not, we should add a debug_assert in `commit_epoch` and `report_compact_task`.

We didn't check it.
just added asserts in compaction